src/Microsoft.AspNet.Mvc.Core/Formatters/InputFormatter.cs

```diff
@@ -126,7 +132,12 @@ protected Encoding SelectCharacterEncoding(MediaTypeHeaderValue contentType)
             }
 
             // No supported encoding was found so there is no way for us to start reading.
-            throw new InvalidOperationException(Resources.FormatInputFormatterNoEncoding(GetType().FullName));
+            return null;
+        }
+
+        protected string GetNoEncodingMessage()
+        {
+            return Resources.FormatInputFormatterNoEncoding(GetType().FullName);
```

Comment:
Yep. I'm sleepy but I realized you need this to not-throw so you can use the validation error instead. 

I actually feel a little bit like just living with duplication of the message is a better choice than the putting this turd here.